sdks/go/container/boot.go

```diff
@@ -99,60 +87,87 @@ func main() {
 		log.Fatalf("Failed to retrieve staged files: %v", err)
 	}
 
-	// TODO(BEAM-13647): Remove legacy hack once aged out.
+	name, err := getGoWorkerArtifactName(artifacts)
+	if err != nil {
+		log.Fatalf("Failed to get Go Worker Artifact Name: %v", err)
+	}
+
+	// (3) The persist dir may be on a noexec volume, so we must
+	// copy the binary to a different location to execute.
+	const prog = "/bin/worker"
+	if err := copyExe(filepath.Join(dir, name), prog); err != nil {
+		log.Fatalf("Failed to copy worker binary: %v", err)
+	}
+
+	args := []string{
+		"--worker=true",
+		"--id=" + *id,
+		"--logging_endpoint=" + *loggingEndpoint,
+		"--control_endpoint=" + *controlEndpoint,
+		"--semi_persist_dir=" + *semiPersistDir,
+		"--options=" + options,
+	}
+	if info.GetStatusEndpoint() != nil {
+		args = append(args, "--status_endpoint="+info.GetStatusEndpoint().GetUrl())
+	}
+
+	log.Fatalf("User program exited: %v", execx.Execute(prog, args...))
+}
+
+func getGoWorkerArtifactName(artifacts []*pipepb.ArtifactInformation) (string, error) {
```

Comment:
This diff is pretty rough 😢 but lines 95-115 are unchanged, most of the stuff in this function was just directly above it